### PR TITLE
feat(amplify_analytics): Error handling + iOS fix

### PR DESCRIPTION
**Pull request by fjnoyp**
*Issue #, if available:*

*Description of changes:*
- Add error handling Dart, Android, iOS
- Fix iOS bug with analytics properties
- Refactor iOS code to support iOS unit tests
- iOS unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_analytics_pinpoint/android/src/main/kotlin/com/amazonaws/amplify/amplify_analytics_pinpoint/AmplifyAnalyticsBridge.kt [hunk 1]
```diff
@@ -33,7 +33,6 @@ class AmplifyAnalyticsBridge {
         private val LOG = AmplifyAnalyticsPinpointPlugin.LOG
 
         fun recordEvent(@NonNull arguments: Any, @NonNull result: MethodChannel.Result) {
-            try {
                 val argumentsMap = arguments as HashMap<*, *>
                 val name = argumentsMap["name"] as String
```
Comment (Amplifiyer) [inline packages/amplify_analytics_pinpoint/android/src/main/kotlin/com/amazonaws/amplify/amplify_analytics_pinpoint/AmplifyAnalyticsBridge.kt:38]: I think the indentation should change as well

File: packages/amplify_analytics_pinpoint/example/ios/unit_tests/AnalyticsUnitTests.swift [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+@testable import amplify_analytics_pinpoint
+
+class AnalyticsUnitTests: XCTestCase {
+
+    var pluginUnderTest: SwiftAmplifyAnalyticsPinpointPlugin = SwiftAmplifyAnalyticsPinpointPlugin()
+
+        
+
+    
+    func test_recordEvent_returnsSuccess() throws {
```
Comment (Amplifiyer) [inline packages/amplify_analytics_pinpoint/example/ios/unit_tests/AnalyticsUnitTests.swift:28]: extra lines

File: packages/amplify_analytics_pinpoint/example/ios/unit_tests/unit_tests.swift [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    func testPerformanceExample() throws {
+        // This is an example of a performance test case.
+        measure {
+            // Put the code you want to measure the time of here.
+        }
+    }
+
+}
```
Comment (Amplifiyer) [inline packages/amplify_analytics_pinpoint/example/ios/unit_tests/unit_tests.swift:32]: should be deleted.

File: packages/amplify_flutter/lib/amplify.dart [hunk 4]
```diff
@@ -184,6 +186,13 @@ class AmplifyClass extends PlatformInterface {
     await DataStore.configure(configuration);
   }
 
+  void _castAndThrowAnalyticsException(PlatformException e) {
+    switch (e.code) {
+      case "AnalyticsException": throw AnalyticsException.fromMap(Map<String, String>.from(e.details));
+      default: throw e;
+    }
+  }
```
Comment (Amplifiyer) [inline packages/amplify_flutter/lib/amplify.dart:194]: We can just throw `AnalyticsException.fromMap(Map<String, String>.from(e.details));` inline

**Review by Amplifiyer — APPROVED** [review]
Great work, looks good

**Review by haverchuck — APPROVED** [review]
LGTM

**Pull request merged**